File: Scheduler/ClientApp/src/app/calendar-models/event-model.ts
import { AppointmentType } from "./appointmentType-enum";
import { RangeType } from "./rangeType-enum";

export class EventModel {
    Id: number = 0;
    Location: string = "";
    Title: string ='';
    Color: string = '';
    Details: string = '';
    OriginalDate: string = '01/01/0001';
    Date: string = '';
    Time: string = '';
    EndDateSpan: number = 0;
    EndTime: string = '';
    YearMonth: string ='';
    IsRepeat: boolean = false;
    IsOwner: boolean = false;
    CreatedBy: string = '';
    CreatorId: number = 0;
    Type: AppointmentType;

    IsDone: boolean = false;
    IsDeleted: boolean = false;
    HasEdit: boolean = false;

    RepeatSelection: number = 0;
    RepeatEnd: number = 0;
    HasNoWeekEnds: boolean = false;

    After: number = 0;
    OnDate: string = '01/01/0001';
    IsClone: boolean = false;
    NumberOfRepeats: number = 0;

    IsRange: boolean = false;
    RangeType: RangeType;

    MainEventReference: EventModel;
    IsEmptyEvent: boolean = false;
    DateId: string = '';
}

File: Scheduler/Models/EventModel.cs
using System;
using Scheduler.Entity;

namespace Scheduler.Models
{
	public class EventModel
	{
		public int Id { get; set; }
		public string Location { get; set; }
		public string Title { get; set; }
		public string Color { get; set; }
		public string Details { get; set; }

		public string ActualYearMonth { get; set; }
		public string YearMonth { get; set; }

		public string OriginalDate { get; set; }
		public string Date { get; set; }

		public string Time { get; set; }
		public bool IsRepeat { get; set; }
        public int CreatorId { get; set; }
		public string CreatedBy { get; set; }
		public bool IsOwner { get; set; }

		public bool IsDeleted { get; set; }
		public AppointmentType Type { get; set; }
		public bool IsDone { get; set; }

		public bool HasEdit { get; set; }

		public RepeatSelectionEnum RepeatSelection { get; set; }
		public RepeatEndEnum RepeatEnd { get; set; }

		public int After { get; set; }
		public string OnDate { get; set; }
		public bool IsClone { get; set; }
		public int NumberOfRepeats { get; set; }
    }
}